src/main/java/com/puppycrawl/tools/checkstyle/SuppressionsStringPrinter.java

```diff
@@ -0,0 +1,152 @@
+                    suppressionLineColumnNumber);
+            throw new IllegalStateException(exceptionMsg);
+        }
+    }
+
+    /**
+     * Extracts line number from config property.
+     * @param suppressionLineColumnNumber line and column number of the suppression.
+     * @return line number.
+     */
+    private static int extractLineNumber(String suppressionLineColumnNumber) {
+        final int colonPosition = suppressionLineColumnNumber.indexOf(COLON_LITERAL);
+        final String lineNumber = suppressionLineColumnNumber.substring(0, colonPosition);
+        return Integer.parseInt(lineNumber);
+    }
+
+    /**
+     * Extracts column number from config property.
+     * @param suppressionLineColumnNumber line and column number of the suppression.
+     * @return column number.
+     */
+    private static int extractColumnNumber(String suppressionLineColumnNumber) {
+        final int colonPosition = suppressionLineColumnNumber.indexOf(COLON_LITERAL);
+        final String columnNumber = suppressionLineColumnNumber.substring(colonPosition + 1);
+        return Integer.parseInt(columnNumber);
+    }
+
+    /**
+     * Parses file and returns {@code DetailAST} object.
+     * @param text {@code FileText} object.
+     * @return {@code DetailAST} object.
+     * @throws CheckstyleException if the file is not a Java source.
+     */
+    private static DetailAST parseFileText(FileText text)
+            throws CheckstyleException {
+        final FileContents contents = new FileContents(text);
+        final DetailAST result;
+        try {
+            result = TreeWalker.parseWithComments(contents);
+        }
+        catch (RecognitionException | TokenStreamException ex) {
+            final String exceptionMsg = String.format(Locale.ROOT,
+                    "%s occurred during the analysis of file %s.",
+                    ex.getClass().getSimpleName(), text.getFile().getPath());
+            throw new CheckstyleException(exceptionMsg, ex);
+        }
+
+        return result;
+    }
+
+    /**
+     * Creates {@code XpathQueryGenerator} instance and generates suppressions.
+     * @param fileText {@code FileText} object.
+     * @param detailAST {@code DetailAST} object.
+     * @param lineNumber line number.
+     * @param columnNumber column number.
+     * @param tabWidth length of the tab character.
+     * @return generated suppressions.
+     */
+    private static String generate(FileText fileText, DetailAST detailAST, int lineNumber,
+                                   int columnNumber, int tabWidth) {
+        final XpathQueryGenerator queryGenerator =
+                new XpathQueryGenerator(detailAST, lineNumber, columnNumber, fileText,
+                        tabWidth);
+        final List<String> suppressions = queryGenerator.generate();
+        final StringBuilder result = new StringBuilder(256);
+        for (int i = 0; i < suppressions.size(); i++) {
```

Comment:
Can we use Stream API from Java 8 here and Collectors.joining as a terminator for the stream? Stream API allows you to simplify the logic and get rid of the conditions. The code without conditions (linear) is easier to read and support.

Reply:
Don't forget stream classes are forbidden because of issues with code coverage.
https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml#L19-L29

Reply:
@rnveach 
I remember. My request was to use stream() -> map -> collect. This pipeline does not require imports from forbidden sources.

Reply:
done